Storage/SQL/SQLiteBookmarksSyncing.swift

```diff
@@ -655,16 +656,29 @@ extension SQLiteBookmarks {
     }
 
     // Retrieve all the local bookmarks that are not present remotely in order to avoid merge logic later.
-    public func getLocalBookmarksAdditions(limit: Int) -> Deferred<Maybe<[BookmarkMirrorItem]>> {
-        let sql =
-        "SELECT * FROM \(TableBookmarksLocal) AS bookmarks " +
-        "WHERE type = \(BookmarkNodeType.bookmark.rawValue) AND sync_status = \(SyncStatus.new.rawValue) AND parentID = ? " +
-        "AND NOT EXISTS (SELECT 1 FROM \(TableBookmarksBuffer) buf WHERE buf.guid = bookmarks.guid) " +
+    public func getLocalBookmarksModifications(limit: Int) -> Deferred<Maybe<(additions: [BookmarkMirrorItem], deletions: [GUID])>> {
+        let deletionsQuery =
+        "SELECT deletions.id FROM \(TablePendingBookmarksDeletions) AS deletions " +
+        "WHERE EXISTS (SELECT 1 FROM \(TableBookmarksBuffer) buf WHERE buf.guid = deletions.id " +
```

Comment:
Do something like this in SQLiteBookmarksSyncing.swift, around line 482:

```
if !deleted.isEmpty {
    for chunk in chunk(deleted, by: BrowserDB.MaxVariableNumber) {
        // … insert usual varlist nonsense here
        "DELETE FROM \(TablePendingBookmarksDeletions) WHERE GUID IN \(varlist)"
    }
}
```

That way there should never be an entry in `TPBD` that doesn't correspond to a non-deleted buffer item.